addons/cookoff/functions/fnc_burnEngine.sqf

```diff
@@ -0,0 +1,37 @@
+/*
+ * Author: KoffeinFlummi, commy2
+ * Start fire in engine block of a car.
+ *
+ * Arguments:
+ * 0: Vehicle <Object>
+ *
+ * Return Value:
+ * Boom.
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_vehicle"];
+
+if (_vehicle getVariable [QGVAR(isEngineSmoking), false]) exitWith {};
+_vehicle setVariable [QGVAR(isEngineSmoking), true, true];
+
+private _position = [
+    0,
+    (boundingBoxReal _vehicle select 1 select 1) - 4,
+    (boundingBoxReal _vehicle select 0 select 2) + 2
+];
+
+private _smoke = "#particlesource" createVehicle [0,0,0];
+_smoke setParticleClass "ObjectDestructionSmoke1_2Smallx";
+_smoke attachTo [_vehicle, _position];
+
+[{
+	(_this select 0) params ["_vehicle", "_smoke", "_time"];
+
+	if (!alive _vehicle || {_vehicle getHitPointDamage "HitEngine" < 0.9} || {CBA_missionTime > _time}) then {
```

Comment:
Maybe you should check isNull _vehicle first, in case e.g. the vehicle was removed by Zeus